### PR TITLE
[Off-chain, Polylog] refactor: replace std logger usage in pkg/... with polylog

### PR DESCRIPTION
**Pull request by bryanchriswhite**
### Human Summary

Refactors usage of go std `log` pkg with analogous usage of polylog in the off-chain (i.e. pkg/...) packages.

## Issue

- #181
- #210 
- #209 
- #211 
- #218

Now that polylog is available, use it instead instead of `log`.

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: pkg/appgateserver/server.go [hunk 1]
```diff
@@ -94,6 +96,13 @@ type appGateServer struct {
 }
 
 // NewAppGateServer creates a new appGateServer instance with the given dependencies.
+//
+// Required dependencies:
```
Comment (Olshansk) [inline pkg/appgateserver/server.go:100]: Noice

File: pkg/partials/partial.go [hunk 2]
```diff
@@ -24,36 +25,40 @@ func GetRequestType(payloadBz []byte) (sharedtypes.RPCType, error) {
 
 // GetErrorReply returns an error reply for the given payload and error,
 // in the correct format required by the request type.
-func GetErrorReply(payloadBz []byte, err error) ([]byte, error) {
-	partialRequest, er := PartiallyUnmarshalRequest(payloadBz)
+func GetErrorReply(ctx context.Context, payloadBz []byte, err error) ([]byte, error) {
```
Comment (Olshansk) [inline pkg/partials/partial.go:28]: It's cool that a result of having a structured logger forced us to propagate the context throughout!

**Review by Olshansk — APPROVED** [review]
Did a review of about half of this PR. I think that it's one of those things we'll iterate on but let's get the base in!

File: pkg/partials/partial.go (hunk 2 above)
Comment (bryanchriswhite) [reply pkg/partials/partial.go:28]: :100:

**Pull request merged**